### ISSUE TITLE
Q: NuGet (Xplat) Server - long term strategy 

### ISSUE THREAD

**Issue by terrajobst** [NuGet/NuGetGallery#3899]
_From @WernerMairl on May 8, 2017 5:11_

Hi
dotnet core is depending (extremely) on Nuget (Protocol, and Gallery)

Now, we have May 2017 and there is no Nuget Server (pure core/xplat) available or at least planned.

For my own/private Nuget Server i must use the full blown (Framework >=4) asp.net and SQL-Server implementation or a Java Implementation (Nexus) or something else.....

All this does not increases my confidence in this system (and i like dotnetcore)

I know that Nuget is a different project and currently not part of the dotnet foundation, but it is also MS-coupled and the Project is really critical for dotnetcore.

What is the plan, what is the strategy?

@terrajobst et.al. ?

regards
Werner

_Copied from original issue: dotnet/core#617_

**Comment by maartenba** [comment]
Afaik NuGet is a .NET Foundation project: https://dotnetfoundation.org/nuget

That said there are tonnes of server implementations out there, from a shared folder (SMB based is possible) over NuGet.Server and the gallery over third party like ProGet and TeamCity and Artifactory over software as a service like AppVeyor and MyGet.

Even a few PHP based implementation (https://hub.docker.com/r/sunside/simple-nuget-server/ on Docker, https://github.com/Daniel15/simple-nuget-server) and probably others.

I believe @emgarten has a static server generator, too.

Not sure if these could alleviate the urgency? (Not meaning to hijack the thread here as long term this probably makes sense)

**Comment by emgarten** [comment]
@WernerMairl would you share a bit more about your scenario?  Are you already using NuGet.Server and want to move to linux only, or are you linux only and trying to switch to dotnet and need a local server?

This is the first request for NuGet.Server on linux that I've seen so far. I think it would be handy, but as @maartenba pointed out, there are already many different ways to host NuGet packages so it has been unclear how much an xplat NuGet.Server would be used.

I typically see go with a 3rd party service for hosting their packages. Internally local folder and file shares are frequently used.

> I believe @emgarten has a static server generator, too.

You can find the static server generator here: https://github.com/emgarten/sleet  It can run purely on netcore on linux, which might make it the closest thing available now to what @WernerMairl is looking for.

**Comment by WernerMairl** [comment]
Never seen sleet before, it looks very nice => my next try!!

No real usecase now, but seeing dotnetcore, kestrel and other cool stuff in dotnetcore, implementing a NuGet Server in pure "core" technology seems to be a musthave in terms of dogfooding.

I see the problems in some used framework inside the "classic server" that seems not be ready for dotnetcore.

This is the reason why i ask more for a plan/strategy then a implementation.

thank you
Werner
PS.: feel free to close the issue any time....

**Comment by devSoheilAlizadeh** [comment]
Any plan for this issue?

**Comment by TylerLeonhardt** [comment]
I would love to see this so that the [PowerShell Gallery](https://www.powershellgallery.com/) (which is a fork of the NuGetGallery) can move to .NET Core as well.

**Comment by joelverhagen** [comment]
No, this work is not planned. For NuGet.Server, we'll resolve critical bug fixes and security issues, but our team is instead focusing on maintaining NuGet.org itself. I recommend one of the following:

1. Use another package source implementation that supports .NET (Core). There is a list of private package sources here: https://learn.microsoft.com/en-us/nuget/hosting-packages/overview. I know that BaGet is .NET Core and several folks have mentioned to me that they're successful with it.
1. Someone in the community that is interested in driving this can submit PR moving the NuGet.Server repository to .NET (Core). I recommend breaking it up into multiple PRs or at least individual reviewable commits if you plan on doing this.

**Comment by Balkoth** [comment]
I ended up using: https://github.com/emgarten/Sleet

**Comment by japarson** [comment]
> [!WARNING]
> **NuGet.Server is deprecated and will no longer receive updates.** No new features, bug fixes, or security patches will be released. The [NuGet.Server package on NuGet.org](https://www.nuget.org/packages/NuGet.Server) has been marked as deprecated. Existing deployments will continue to function, but running deprecated and unpatched software in production poses a security risk. We encourage all users to migrate to an alternative solution such as [Azure Artifacts](https://azure.microsoft.com/en-us/products/devops/artifacts/) or [GitHub Packages](https://github.com/features/packages) as soon as possible.
